packages/neos-ui-editors/src/Editors/DateTime/helpers.ts

```diff
@@ -33,16 +35,29 @@ export default function (format) {
     };
     return format.replace(
         /[dDjlNwWFmMnoYyaAgGhHisOPU]/g,
+        // @ts-ignore
         phpStr => formatMap[phpStr]
     );
 }
 
 //
 // Check if given format has a date formatting
 //
-export const hasDateFormat = format => /[yYFmMntdDjlNSw]/.test(format);
+export const hasDateFormat = (format: string) => /[yYFmMntdDjlNSw]/.test(format);
 
 //
 // Check if given format has a time formatting
 //
-export const hasTimeFormat = format => /[gGhHis]/.test(format);
+export const hasTimeFormat = (format: string) => /[gGhHis]/.test(format);
+
+export const tryDateFrom = (value: string | Date | undefined | null) => {
```

Comment:
This is a weird function name.

Reply:
the real thing which is weird is that we allow Date itself as input

And I myself am still in favour of tryDateFromString and handle non string inputs beforehand

Reply:
Slightly better. But don't you `createDateFromString`?

Reply:
The From without suffix is basically an implicit fromAny or fromMixed


The reason we use the prefix try is because the return value is not guaranteed to be a Date. if we would leave that out I would expect an exception in case of a failure (what we don’t want/need)

Reply:
I agree using create as prefix in general is a good practice.

the reason why I dindt use it because I don’t include it as prefix as static factories in php dot‘s either and I thought the php class would correlate kind of with this JavaScript module file „helper.js“. This assumption is of course not true as the module should the be called „date.js“ and must hold an actual dto which will be returned and other exports that operate on the dto.

so I agree on `tryCreateDateFromString`